### ISSUE TITLE
Problem with "Waiting for this message. This may take a while."

### ISSUE THREAD

**Issue by thejockel** [tgalal/yowsup#1782]
Hey,
I have a Problem with theses kinds of messages in a group chat using the demo client.

Can someone tell me how to avoid theses?

<img width="612" alt="screen shot" src="https://cloud.githubusercontent.com/assets/12292135/18942442/60664ee0-8619-11e6-8ba8-1e783bfd893d.png">

Thx Jockel

**Comment by jlguardi** [comment]
Don't delete your axolotl.db, keep updated recipient keys and don't ignore retry acks.

**Comment by thejockel** [comment]
Hey thanks for your replay.
I am using the yowsup-cli demos client and do not delete the axolotl.db.

**Comment by jlguardi** [comment]
I'll describe an unmanaged situation:
 - You (yowsup) get recipient (official) keys
 - recipient reinstalls and so, his keys change
 - you send a message with old keys
 - you get simple check
 - you get ack retry instead of double check due to decrypt error
 - axolotl layer doesn't have the original message (this layer doesn't have any persistence) so retry is sent to upper layers.
 - by default, the client layer just prints the issue but no action is taken. 
 This happends 4-5 times and then the reported err message appears in the recipient side. 

Solution: implement persistence a management I the axolotl layer.

**Comment by sidnet03** [comment]
Hi,

Is there a solution for this? I sent a message from my rpi3 to myself and it says delivered. I got a chat on my cell but it says "waiting for this message. This may take a while". It's been 15 mins now - how to i solve this?

Regards,
Sid

**Comment by zuhannes** [comment]
hi jlguardi,

Referring your 2nd october comments, how do i actually do this on the yowsup-cli program?.

**Comment by zuhannes** [comment]
I just want to update i solve my problem of waiting for my message on yowsup-cli by supplying -m on command line. Now every message with yowsup-cli reach the chat group without error

**Comment by pippopippo69** [comment]
hi , same problem
I do not know how to "implement persistence a management I the axolotl layer"..

**Comment by atimothee** [comment]
he's how i solved my problem, i put the bot number in a physical device, and logged into whatsapp, then i removed it, and started a new bot on a different server.

**Comment by bbtimx** [comment]
I have this problem too.

Everything is working fine until an new group member is invited.

I´ve implemented 

```
jids = [self.aliasToJid(phone) for phone in phone.split(',')]
entity = GetKeysIqProtocolEntity(jids)
```

but this don´t work.

How I implement fetching the ne group key(s) before sending a message to the group?

**Comment by SPWC** [comment]
yes how do I implement fetching the new group key(s)/key(s) before sending a message to the group or contact?

**Comment by mocart2** [comment]
+1 i have same problem, tryed to use -m key, but it not helps, maybe anyone have solution?
